src/tool/subcommands/api_cmd.rs

```diff
@@ -1877,15 +1881,36 @@ async fn run_tests(
             }
             _ => false,
         };
+        if success {
+            if let Some(dump_dir) = &dump_dir {
+                if let Some(test_dump) = &test_result.test_dump {
+                    let dir = dump_dir.join(if success { "valid" } else { "invalid" });
```

Comment:
we already checked `if success` before so error `invalid` is unreachable right

Reply:
Ah good catch, I should have removed the outer check

Reply:
Fixed.